app/src/main/java/org/mozilla/fenix/crashes/CrashContentInteractor.kt

```diff
@@ -0,0 +1,88 @@
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import org.mozilla.fenix.components.AppStore
+import org.mozilla.fenix.components.Components
+import org.mozilla.fenix.components.appstate.AppAction
+import org.mozilla.fenix.utils.Settings
+
+/**
+ * Handles all user interactions with a [CrashContentView].
+ */
+class CrashContentInteractor(
+    private val sessionId: String,
+    private val components: Components,
+    private val settings: Settings,
+    private val appStore: AppStore
+) {
+
+    /**
+     * Restore all sessions and optionally report pending non-fatal crashes.
+     *
+     * @param reportCrashes If true, submit crash reports for all current non-fatal crashes.
+     * @return [Job] allowing to check status / cancel the reporting operation or null if reporting is disabled.
+     */
+    fun handleCloseAndRestore(reportCrashes: Boolean): Job? {
+        val job = maybeSubmitPendingNonFatalCrashes(reportCrashes)
+
+        components.useCases.sessionUseCases.crashRecovery.invoke()
+
+        return job
+    }
+
+    /**
+     * Closes the current tab, restore all sessions and optionally report pending non-fatal crashes.
+     *
+     * @param reportCrashes If true, submit crash reports for all current non-fatal crashes.
+     * @return [Job] allowing to check status / cancel the reporting operation or null if reporting is disabled.
+     */
+    fun handleCloseAndRemove(reportCrashes: Boolean): Job? {
+        val job = maybeSubmitPendingNonFatalCrashes(reportCrashes)
+
+        components.useCases.tabsUseCases.removeTab(sessionId)
+        components.useCases.sessionUseCases.crashRecovery.invoke()
+
+        return job
+    }
+
+    /**
+     * Submits all pending non-fatal crash reports if the "Send crash" checkbox was checked
+     * and the report crashes setting is enabled.
+     * Also clears the current list of non-fatal crashes irrespective of whether they are reported or not.
+     *
+     * @param reportCrashes A second condition beside crash reporting being enabled in app settings
+     * based on which the current crashes will be reported or immediately disposed off.
+     * @return [Job] allowing to check status / cancel the reporting operation or null if reporting is disabled.
+     */
+    @VisibleForTesting
+    @OptIn(DelicateCoroutinesApi::class) // GlobalScope usage
+    internal fun maybeSubmitPendingNonFatalCrashes(reportCrashes: Boolean): Job? {
+        var job: Job? = null
+        if (reportCrashes && settings.isCrashReportingEnabled) {
+            job = GlobalScope.launch(Dispatchers.IO) {
+                val crashes = appStore.state.nonFatalCrashes
+                crashes.forEach {
+                    components.analytics.crashReporter.submitReport(it)
+                    appStore.dispatch(AppAction.RemoveNonFatalCrash(it))
+
+                    Log.i("Crash Reporter", "Report \"${it.uuid}\" submitted.")
```

Comment:
do we need these logs? Can we add an equivalent `Logger('CrashContentInteracator')`?

Reply:
Removed them.